src/did/methods/eth_resolver.js

```diff
@@ -0,0 +1,24 @@
+import { isValidAddress } from 'ethereumjs-util';
+import { parse } from '../parse';
+
+const generateDocument = (did, address) => ({
+	'@context': 'https://www.w3.org/2019/did/v1',
+	id: did,
+	publicKey: [
+		{
+			id: `${did}#keys-1`,
+			type: 'Secp256k1VerificationKey2018',
+			controller: did,
+			ethereumAddress: address
+		}
+	],
+	authentication: [`${did}#keys-1`]
+});
+
+export const resolve = async did => {
+	const { method, idString } = parse(did);
+	if (method !== 'eth' || !isValidAddress(idString)) {
+		return Promise.reject(new Error('Not a valid eth DID'));
```

Comment:
Better to use `return` and `throw` instead of `Promise.resolve` and `Promise.reject` accordingly

Reply:
Ooops, I guess old habits die hard. You are right, done.